### PR TITLE
APM guidance

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
#### What is this PR About?

APM guidance

I know there is a bit of rush on this document. I couldn't test it because the test container doesn't start. Likely docker 1.13 is broken in fedora 27.

Please if someone has time to review it do a deep  review.

cc: @redhat-cop/cant-contain-this

### CODE REVIEW

**Comment by raffaelespazzoli** [comment]
@etsauer I reviewed the APM page, it should be fine now. The Travis tests are failing for reasons that are unclear to me. Please advise.

**Comment by etsauer** [comment]
the CI build is getting SSL errors on the links in this doc, which I am unable to replicate.
```- ./_site/playbooks/app_dev/APM_guidance.html
  *  External link https://prometheus.io failed: response code 0 means something's wrong.
             It's possible libcurl couldn't connect to the server or perhaps the request timed out.
             Sometimes, making too many requests at once also breaks things.
             Either way, the return message (if any) from the server is: SSL connect error
- ./_site/playbooks/continuous_delivery/external-jenkins-integration.html
  *  External link https://jenkins.io failed: response code 0 means something's wrong.
             It's possible libcurl couldn't connect to the server or perhaps the request timed out.
             Sometimes, making too many requests at once also breaks things.
             Either way, the return message (if any) from the server is: SSL connect error
htmlproofer 3.7.5 | Error:  HTML-Proofer found 2 failures!
```

@sabre1041 @JaredBurck any ideas?

**Comment by raffaelespazzoli** [comment]
@etsauer done.

**Comment by etsauer** [comment]
@raffaelespazzoli will you rebase your branch please? We fixed the CI build in another PR.

File: playbooks/app_dev/APM_guidance.adoc [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+Note that in this case very little additional work is required in order to enable APM and that no additional image configuration is needed.
+
+=== Market players
+
+
+Major market players in APM are:
+
+ * https://www.dynatrace.com/[Dynatrace]
+ * https://www.appdynamics.com/[appDynamics]
```
Comment (JaredBurck) [inline playbooks/app_dev/APM_guidance.adoc]: Capitalize 'A' in appDynamics

File: playbooks/app_dev/APM_guidance.adoc [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+https://sysdig.com/openshift-monitoring/[https://sysdig.com/openshift-monitoring/] +
+https://support.sysdig.com/hc/en-us/articles/211421063-Sysdig-Install-OpenShift[https://support.sysdig.com/hc/en-us/articles/211421063-Sysdig-Install-OpenShift] +
+https://blog.openshift.com/openshift-ecosystem-using-sysdig-monitor-openshift/[https://blog.openshift.com/openshift-ecosystem-using-sysdig-monitor-openshift/]
+
+==== Installing Instana on OpenShift
+
+https://blog.openshift.com/deploying-instana-apm-natively-openshift-environment/[https://blog.openshift.com/deploying-instana-apm-natively-openshift-environment/] +
+https://instana.atlassian.net/wiki/spaces/DOCS/pages/15630376/Docker+scheduled#Docker(scheduled)-OpenShift[https://instana.atlassian.net/wiki/spaces/DOCS/pages/15630376/Docker+scheduled#Docker(scheduled)-OpenShift]
+
+== Distributed Tracing
```
Comment (JaredBurck) [inline playbooks/app_dev/APM_guidance.adoc]: You might also want to mention tools such as Zipkin and NetFlix OSS.  The Red Hat Developers MSA [1] also has a demo of this with OpenTracing, Zipkin, and Jaeger.

[1] https://github.com/redhat-helloworld-msa/helloworld-msa

File: playbooks/app_dev/APM_guidance.adoc [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+A jaeger client (http://jaeger.readthedocs.io/en/latest/client_libraries/[client libraries] for various languages exist) will generate trace information and sends them via a local UDP to a jaeger agent, running as a http://blog.kubernetes.io/2015/06/the-distributed-system-toolkit-patterns.html[sidecar]. The jaeger agent will perform the necessary sampling and throttling, and then send the traces to the central jaeger collector, which stores them in http://cassandra.apache.org/[Cassandra]. Once the traces are stored they can be visualized using the UI.
+
+Instructions on how to install jaeger in OpenShift can be found https://github.com/jaegertracing/jaeger-openshift[here].
+
+The UI presents result as follows:
+
+image::images/jaeger_ui.png[]
+
+If you are building a http://blog.christianposta.com/microservices/application-network-functions-with-esbs-api-management-and-now-service-mesh/[service mesh] with https://istio.io/[istio], consider that istio is naturally integrated with jaeger, as explained https://istio.io/docs/tasks/telemetry/distributed-tracing.html[here].
```
Comment (JaredBurck) [inline playbooks/app_dev/APM_guidance.adoc]: Istio OpenShift reference links:
[1] https://blog.openshift.com/red-hat-istio-launch/
[2] https://blog.openshift.com/running-istio-service-mesh-openshift/
[3] https://blog.openshift.com/evaluate-istio-openshift/
[4] https://blog.openshift.com/istio-traffic-management-diving-deeper/
[5] https://blog.openshift.com/openshift-commons-briefing-103-microservices-istio-openshift-chak-jonagam-red-hat/

File: playbooks/app_dev/APM_guidance.adoc [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+In this approach the agents are deployed as https://docs.openshift.com/container-platform/latest/dev_guide/daemonsets.html[daemon sets]. The containers in this daemon set are privileged and can inspect the other containers running on the same node. Once a new container starts, the agent sends all available information of that container to the metrics collector.
+
+image::images/APMGuidance-ContainerAPMArch2.png[]
+
+Note that in this case very little additional work is required in order to enable APM and that no additional image configuration is needed.
+
+=== Market players
+
+
+Major market players in APM are:
```
Comment (lpsantil) [inline playbooks/app_dev/APM_guidance.adoc]: Mention OpenShift Primed here:

https://hub.openshift.com/primed

File: playbooks/app_dev/APM_guidance.adoc (hunk 1 above)
Comment (raffaelespazzoli) [reply playbooks/app_dev/APM_guidance.adoc]: done

File: playbooks/app_dev/APM_guidance.adoc (hunk 2 above)
Comment (raffaelespazzoli) [reply playbooks/app_dev/APM_guidance.adoc]: I added the reference to hello world msa, I don't think that zipkin should be mentioned as it is obsoleted by Jaeger.

File: playbooks/app_dev/APM_guidance.adoc (hunk 3 above)
Comment (raffaelespazzoli) [reply playbooks/app_dev/APM_guidance.adoc]: what should I do with these links? This article is not about istio.

File: playbooks/app_dev/APM_guidance.adoc (hunk 4 above)
Comment (raffaelespazzoli) [reply playbooks/app_dev/APM_guidance.adoc]: Can you explain where exactly you would like to see this link and why?

Comment (lpsantil) [reply playbooks/app_dev/APM_guidance.adoc]: Something like

[OpenShift Primed] lists several market players.  Below are some links to their APM offerings for OpenShift.

Comment (raffaelespazzoli) [reply playbooks/app_dev/APM_guidance.adoc]: added a consideration on primed

**Comment by raffaelespazzoli** [comment]
@etsauer I have incorporated the feedback, I think we can merge now.

File: playbooks/app_dev/APM_guidance.adoc [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+A jaeger client (http://jaeger.readthedocs.io/en/latest/client_libraries/[client libraries] for various languages exist) will generate trace information and sends them via a local UDP to a jaeger agent, running as a http://blog.kubernetes.io/2015/06/the-distributed-system-toolkit-patterns.html[sidecar]. The jaeger agent will perform the necessary sampling and throttling, and then send the traces to the central jaeger collector, which stores them in http://cassandra.apache.org/[Cassandra]. Once the traces are stored they can be visualized using the UI.
+
+Instructions on how to install jaeger in OpenShift can be found https://github.com/jaegertracing/jaeger-openshift[here].
+
+The UI presents result as follows:
+
+image::images/jaeger_ui.png[]
+
+https://github.com/redhat-helloworld-msa/helloworld-msa[Helloworld-MSA] is an example of how to instrument an application with Jaeger.
```
Comment (etsauer) [inline playbooks/app_dev/APM_guidance.adoc]: this link is formatted incorrectly and is therefore breaking the build.

**Pull request merged**